sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/backoff/ExponentialBackOff.java

```diff
@@ -0,0 +1,101 @@
+package com.mesosphere.sdk.scheduler.plan.backoff;
+
+import com.mesosphere.sdk.offer.CommonIdUtils;
+import com.mesosphere.sdk.offer.LoggingUtils;
+import com.mesosphere.sdk.offer.TaskException;
+
+import org.apache.mesos.Protos;
+import org.slf4j.Logger;
+
+import java.time.Duration;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * TODO.
+ */
+public final class ExponentialBackOff {
+
+  private static final Object lock = new Object();
+
+  private static volatile ExponentialBackOff instance;
+
+  private ConcurrentHashMap<String, Delay> delays;
+
+  private double backoffFactor;
+
+  private Duration initialBackoff;
+
+  private Duration maxLaunchDelay;
+
+  private Logger logger;
+
+  private ExponentialBackOff(
+          double backoffFactor,
+          long backoffSeconds,
+          long maxLaunchDelaySeconds)
+  {
+    this.logger = LoggingUtils.getLogger(ExponentialBackOff.class);
+    this.backoffFactor = backoffFactor;
+    this.initialBackoff = Duration.ofSeconds(backoffSeconds);
+    this.maxLaunchDelay = Duration.ofSeconds(maxLaunchDelaySeconds);
+    delays = new ConcurrentHashMap<>();
+  }
+
+  public static ExponentialBackOff getInstance() {
+    if (instance == null) {
+      synchronized (lock) {
+        if (instance == null) {
+          // SUPPRESS CHECKSTYLE MagicNumberCheck
+          instance = new ExponentialBackOff(1.5, 60, 300);
```

Comment:
We'll likely need to parametrize this down the line, this is fine for investigation.